### PR TITLE
Don't select development appliances on AT PMT change

### LINKED ISSUE #135: Http mapping for appliance started in dev mode are added when adding PMT into AT
Scenario:
  * start appliance in development mode
  * add PMT into appliance source appliance type

As a result http mapping will be added into development appliance. Such http mapping should be added only into appliances started in production mode, where PMT modification is not possible.

This issue causes problems wil clew serialization.

### PR DESCRIPTION
**Pull request by mkasztelnik**
When `PortMappingTemplate` is added, removed from `ApplianceType` only `Appliances` started in production should be affected. Development `Appliances` have PMT copied into `DevModePropertySet` and should not be modified.

Fix #135

/cc @dice-cyfronet/atmo-dev-team please take a look. If you have better way how to select only production appliances please let me know.

### CODE REVIEW

**Comment by tbartynski** [comment]
:+1:

**Comment by mkasztelnik** [comment]
@nowakowski, @Nuanda any other volunteer to review this change? It is quite serious bug which should be deployed into production ASAP.

**Comment by Nuanda** [comment]
I checked that but decided to let @nowakowski see that SQL query :).

**Comment by mkasztelnik** [comment]
:ok_hand: - thx, so we are waiting for @nowakowski feedback

File: spec/queries/atmosphere/appliances_affected_by_pmt_spec.rb [hunk 1]
```diff
@@ -30,10 +32,10 @@
...
     it 'finds affected appliances using dev_mode_property_set relation' do
-      dev_as = create(:dev_appliance_set)
-      appl1 = create(:appliance, appliance_type: at, appliance_set: dev_as)
-      create(:appliance, appliance_type: at)
+      appl1 = dev_appliance
       dev_pmt = appl1.dev_mode_property_set.port_mapping_templates.first
+      # other appliances which should not be taken into account
+      create(:appliance, appliance_type: at)
 
       affected_appl = Atmosphere::AppliancesAffectedByPmt.new(dev_pmt).find
```
Comment (nowakowski) [inline spec/queries/atmosphere/appliances_affected_by_pmt_spec.rb]: This looks funky. Wouldn't it make more sense to implement `find` as a static method of `AppliancesAffectedByPmt` (possibly renaming the class to something like `AffectedAppliancesLocalizer`) and parameterize this method with `dev_pmt`?

ATM this invocation reads like you're "creating" some sort of "appliance affected by PMT" and then invoking something on it. I understand what's really going on, but the choice of names is rather unfortunate...

**Comment by nowakowski** [comment]
Query is kosher; not sure I like the declaration of `AppliancesAffectedByPmt.find` (see inline comment).

File: spec/queries/atmosphere/appliances_affected_by_pmt_spec.rb (hunk 1 above)
Comment (mkasztelnik) [inline spec/queries/atmosphere/appliances_affected_by_pmt_spec.rb]: This is an implementation of Query Object taken [from this article](http://blog.codeclimate.com/blog/2012/10/17/7-ways-to-decompose-fat-activerecord-models/) which was created several months ago... Please in the scope of this PR focus on introduced changes. If you believe that name or location of the method is incorrect please do the change and submit another PR :smiley_cat:

**Comment by mkasztelnik** [comment]
@nowakowski please see my response into your previous comment. Can we focus here on introduced changes?

**Comment by nowakowski** [comment]
@mkasztelnik http://bit.ly/1aVUQ1F

**Comment by nowakowski** [comment]
:+1:

**Comment by mkasztelnik** [comment]
:green_heart:
